### ISSUE TITLE
UnusedPrivateMember and fun interfaces

### ISSUE THREAD

**Issue by sanyarnd** [detekt/detekt#5126]
I'm not really sure if it must be considered a bug, but I have the following code.

```kotlin
fun interface A {
    fun doSomething(a: Int, b: Int)
}

fun f(g: A) {
    g.doSomething(1, 2)
}

fun a(a: Int, b: Int) { ... } // <-- reports unused variable

val x: Unit = f(::a)
```

As you can see, function `a` cannot omit arguments, it'll break types.
On the other hand, I'm not sure if it's possible to detect such cases.

**Comment by BraisGabin** [comment]
You could code it like this:

```kotlin
fun interface A {
    fun doSomething(a: Int, b: Int)
}

fun f(g: A) {
    g.doSomething(1, 2)
}

fun a() { ... }

val x: Unit = f { _, _ -> a()}
```

I understand that you want to ues the `::a` notation and that this is just a sample and the real code could be way more complex. But If we allow this case we should not report any function parameter because of this.

I feel that this is not a false-positive bebause with a little change in your code you can fix the unused parameter. And I imagine that this is not something common in your code base so maybe an `@Suppress` have sense in this scenario.

A third option would be to have a flag in this rule to enable/disable the check of function parameters. But I think that we shouldn't go in that direction.

**Comment by sanyarnd** [comment]
My use case includes a function with 9 arguments.

I can rewrite it as
```kotlin
val x: SomeParametrizedInterface<Type1, Type2, Type3> = SomeParametrizedInterface { argument1, anotherArgument, somethingElse, .... -> 
}
```
but as you can see, it'll create a very long line and it'll fail on max line length check x)

I think too that `@Suppress` is a good candidate for such cases, I just wanted to highlight the issue, perhaps someone from dev team has a better idea.

**Comment by schalkms** [comment]
Closing this then due to the arguments given in 
https://github.com/detekt/detekt/issues/5126#issuecomment-1193280299
